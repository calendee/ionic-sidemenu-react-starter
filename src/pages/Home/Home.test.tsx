import React from 'react';
import { renderWithUser, wait } from '../../../test/test-utils';

import HomePage from './Home';

describe('Home Page', () => {
  test('renders home page with a user', async () => {
    const user = { firstName: 'Jane', lastName: 'Doe', uid: 'XYZ123' };
    const { container, getByText } = renderWithUser(<HomePage />, { user });
    await wait();
    getByText(/jane doe/i);
    expect(container).toMatchSnapshot();
  });
});
