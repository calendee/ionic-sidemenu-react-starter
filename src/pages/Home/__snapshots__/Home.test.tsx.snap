// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page renders home page with a user 1`] = `
<div>
  <div
    class="ion-page"
    data-testid="home-page"
  >
    <ion-header>
      <ion-toolbar>
        <ion-buttons>
          <ion-menu-button />
        </ion-buttons>
        <ion-title>
          Home : Jane Doe
        </ion-title>
      </ion-toolbar>
    </ion-header>
    <ion-content>
      <ion-card
        class="welcome-card"
        classname="welcome-card"
      >
        <img
          alt=""
          src="/assets/shapes.svg"
        />
        <ion-card-header>
          <ion-card-subtitle>
            Get Started
          </ion-card-subtitle>
          <ion-card-title>
            Welcome to Ionic
          </ion-card-title>
        </ion-card-header>
        <ion-card-content>
          <p>
            Now that your app has been created, you'll want to start building out features and components. Check out some of the resources below for next steps.
          </p>
        </ion-card-content>
      </ion-card>
      <ion-grid>
        <ion-row>
          <ion-col>
            <ion-button>
              Click Me!
            </ion-button>
          </ion-col>
          <ion-col>
            <ion-button>
              Click Me Too!
            </ion-button>
          </ion-col>
        </ion-row>
      </ion-grid>
      <ion-list>
        <ion-list-header>
          <ion-label>
            Resources
          </ion-label>
        </ion-list-header>
        <ion-item>
          <ion-icon />
          <ion-label>
            Ionic Documentation
          </ion-label>
        </ion-item>
        <ion-item>
          <ion-icon />
          <ion-label>
            Scaffold Out Your App
          </ion-label>
        </ion-item>
        <ion-item>
          <ion-icon />
          <ion-label>
            Change Your App Layout
          </ion-label>
        </ion-item>
        <ion-item>
          <ion-icon />
          <ion-label>
            Theme Your App
          </ion-label>
        </ion-item>
      </ion-list>
    </ion-content>
  </div>
</div>
`;
